onnx/backend/test/case/node/negativeloglikelihoodloss.py

```diff
@@ -21,16 +21,31 @@ def compute_negative_log_likelihood_loss(input, target, weight=None, reduction='
         for i in range(N):
             if target[i] != ignore_index:
                 neg_gather_element_input[i] = -input[i][target[i]]
-    else:
+    elif len(input_shape) == 3:
+        N, C, dim1 = input_shape
+        neg_gather_element_input = np.zeros((N, dim1), dtype=np.float32)
+        for i in range(N):
+            for d1 in range(dim1):
+                if target[i][d1] != ignore_index:
+                    neg_gather_element_input[i][d1] = -input[i][target[i][d1]][d1]
+    elif len(input_shape) == 4:
         N, C, dim1, dim2 = input_shape
         neg_gather_element_input = np.zeros((N, dim1, dim2), dtype=np.float32)
         for i in range(N):
             for d1 in range(dim1):
                 for d2 in range(dim2):
                     if target[i][d1][d2] != ignore_index:
                         neg_gather_element_input[i][d1][d2] = -input[i][target[i][d1][d2]][d1][d2]
+    else:
+        raise NotImplementedError
 
     loss = neg_gather_element_input
+
+    if weight is None and ignore_index is not None:
+        c = input_shape[1]
+        weight = np.ones(c, dtype=np.float32)
+        weight[ignore_index] = np.float32(0)  # type: ignore
```

Comment:
weight[ignore_index] sometime may not be valid. `ignore_idex` can be 100 while the weight is just a 10-element vector.